core/src/main/java/org/apache/iceberg/avro/GenericAvroReader.java

```diff
@@ -155,6 +162,41 @@ public ValueReader<?> record(Type partner, Schema record, List<ValueReader<?>> f
       return recordReader(readPlan, avroSchemas.get(partner), record.getFullName());
     }
 
+    Object toGenericRecord(Type type, Object data) {
+      // Recursively convert data to GenericRecord if type is a StructType.
+      // TODO: Rewrite this as a visitor.
+      if (type instanceof Types.StructType) {
+        Types.StructType structType = (Types.StructType) type;
+        GenericData.Record genericRecord = new GenericData.Record(AvroSchemaUtil.convert(type));
+        int index = 0;
+        for (Types.NestedField field : structType.fields()) {
+          genericRecord.put(
+              field.name(), toGenericRecord(field.type(), ((GenericRecord) data).get(index)));
+          index++;
```

Comment:
Here we assume that the structs are in the same position. For example:
```
struct<a (1): int, b (2): string>
```

But they could be swapped:
```
struct<b (2): string, a (1): int>
```

Reply:
This will be converting actual data (some constant) coming from the schema. The data is represented in a JSON representation denoting the default value. Not sure if there will be a notion of field ID here. Further, since this is converting data from one representation to another, the target object is created from scratch and we are just reformatting the input object so not sure if field ID comparison applies here.